### DISCUSSION TITLE
Controlled components and onChange callback

### DISCUSSION THREAD

**Discussion by Lighttree** [adobe/react-spectrum#4455] [Q&A]
Hi guys, I have a generic question about "Controlled" components. I've noticed that `useControlledState` doesn't call the `onChange` callback when "controlled" behavior is enabled.

Is this expected behavior for controlled components in general? My expectations were that the "controlled" component should call the `onChange` event if the value changes.

I've created a simple test to show:
https://codesandbox.io/s/flamboyant-wildflower-l0ckvp?file=/src/switch.test.js

See how `<Switch />` doesn't call the `onChange` event when I change the `isSelected` prop.

Is this expected?

**Comment by snowystinger** [comment]
I think you forgot to save the codesandbox.

Just to clarify, in a controlled state, ie, you pass the value prop to a component, we will not call on change in response to you changing the prop. Instead, we call on change when we think you should update your prop.


https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components
